Sources/Router/OARoutingHelper.mm

```diff
@@ -617,6 +622,11 @@ - (BOOL) updateCurrentRouteStatus:(CLLocation *)currentLocation posTolerance:(fl
 
 - (CLLocation *) setCurrentLocation:(CLLocation *)currentLocation returnUpdatedLocation:(BOOL)returnUpdatedLocation previousRoute:(OARouteCalculationResult *)previousRoute targetPointsChanged:(BOOL)targetPointsChanged
 {
+    if (![currentLocation hasBearing] && _app.locationServices.lastKnownHeading > 0)
+    {
+        currentLocation = [[CLLocation alloc] initWithCoordinate:currentLocation.coordinate altitude:currentLocation.altitude horizontalAccuracy:currentLocation.horizontalAccuracy verticalAccuracy:currentLocation.verticalAccuracy course:_app.locationServices.lastKnownHeading speed:currentLocation.speed timestamp:currentLocation.timestamp];
+    }
+    
```

Comment:
Remove it. It should not impact when approx bearing is active. Other cases - other issue. Also course is not heading - should be fixed another way.